backend/api.test/DbContextTestSetup.cs

```diff
@@ -54,4 +64,104 @@ public class DatabaseCollection : ICollectionFixture<DatabaseFixture>
+        {
+        }
+
+        protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+        {
+            var claims = new[] { new Claim(ClaimTypes.Name, "Test.User"), new Claim(ClaimTypes.Role, "Test.User") };
+            var identity = new ClaimsIdentity(claims, AuthenticationScheme);
+            var principal = new ClaimsPrincipal(identity);
+            var ticket = new AuthenticationTicket(principal, AuthenticationScheme);
+
+            var result = AuthenticateResult.Success(ticket);
+
+            return Task.FromResult(result);
+        }
+    }
+
+
+    public class MockEchoService : IEchoService
+    {
+        public EchoMission MockEchoMission = new()
+        {
+            Id = 1,
+            Name = "test",
+            URL = new Uri("https://www.I-am-echo-stid-tag-url.com"),
+            Tags = new List<EchoTag>()
+        };
+        public async Task<IList<EchoMission>> GetMissions(string? installationCode)
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return new List<EchoMission>(new EchoMission[] { MockEchoMission });
+        }
+
+        public async Task<EchoMission> GetMissionById(int missionId)
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return MockEchoMission;
+        }
+
+        public async Task<IList<EchoPlantInfo>> GetEchoPlantInfos()
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return new List<EchoPlantInfo>();
+        }
+    }
+
+    public class MockIsarService : IIsarService
+    {
+        public async Task<IsarServiceStartMissionResponse> StartMission(
+            Robot robot,
+            int echoMissionId,
+            IsarMissionDefinition missionDefinition
+        )
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            var isarServiceStartMissionResponse = new IsarServiceStartMissionResponse(
+                isarMissionId: "test",
+                startTime: DateTimeOffset.UtcNow,
+                tasks: new List<IsarTask>()
+            );
+            return isarServiceStartMissionResponse;
+        }
+
+        public async Task<IsarStopMissionResponse> StopMission(Robot robot)
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return new IsarStopMissionResponse();
+        }
+
+        public async Task<HttpResponseMessage> PauseMission(Robot robot)
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return new HttpResponseMessage();
+        }
+
+        public async Task<HttpResponseMessage> ResumeMission(Robot robot)
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return new HttpResponseMessage();
+        }
+
+        public async Task<IsarMissionDefinition> GetIsarMissionDefinition(EchoMission echoMission)
+        {
+            await Task.Run(() => Thread.Sleep(1));
+            return new IsarMissionDefinition(new List<IsarTaskDefinition>());
+        }
+    }
```

Comment:
These classes should be in their own files, maybe in a new "mocks" folder

Reply:
Agreed